src/argilla/tasks/datasets/list.py

```diff
@@ -0,0 +1,73 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from enum import Enum
+from typing import Dict, Optional
+
+import typer
+
+
+class DatasetKind(str, Enum):
+    feedback = "feedback"
+    other = "other"
+
+
+def list_datasets(
+    workspace: Optional[str] = typer.Option(None, help="List datasets in this workspace"),
+    kind: Optional[DatasetKind] = typer.Option(
+        None,
+        help="List datasets of this kind. This option can be used multiple times. By default, all datasets are listed.",
+    ),
```

Comment:
I assume you used the `kind` keyword arg to not overwrite the default `type`, but I think we can name this `dataset_type` instead or even `type_` if applicable

```suggestion
    dataset_type: Optional[DatasetType] = typer.Option(
        None,
        help="List datasets of this type. This option can be used multiple times. By default, all datasets of any type are listed.",
    ),
```

Reply:
Or alternatively, you can make it simpler and define the `typer.Option` custom name as `--type` shortened to `-t` if you prefer to avoid users from typing `--dataset-type` instead